### ISSUE TITLE
Successively disabling and enabling inlay hints doesn't reliably work without typing a character

### ISSUE THREAD

**Issue by x-hgg-x** [rust-lang/rust-analyzer#12027]
**rust-analyzer version**: rust-analyzer version: 65fbe0a8d 2022-04-18 stable

**rustc version**: rustc 1.60.0 (7737e0b5c 2022-04-04)

**relevant settings**: Vscode 1.66 on Linux

In the previous version of the extension, I could easily toogle inlay hints with the `Rust Analyzer: Toggle inlay hints` command.
Now, with the switch to LSP inlay hints, I cannot reenable inlay hints immediately after disabling it, and I need to type a character to "refresh" rust-analyzer.

In the following video, the inlay hints stop working at `0:06`, then after I insert a space at `0:10`, it works again at `0:12`:

https://user-images.githubusercontent.com/39058530/163887619-07ec79c3-4fc2-4ef6-966e-4935f7b97cff.mp4

**Comment by lnicola** [comment]
I was ready to chalk this off as a Code bug, but the command works in a TypeScript file.

**Comment by SE2Dev** [comment]
Toggle Inlay hints doesn't appear to respect `"rust-analyzer.inlayHints.enable": false,` in the latest update. My current workaround is to disable inlay hints for Rust entirely via  ```json
    "[rust]": {
        "editor.inlayHints.enabled": false
    },
    ```

**Comment by lnicola** [comment]
There's no more `rust-analyzer.inlayHints.enable` in the latest update, `Toggle inlay hints` toggles `editor.inlayHints.enabled` instead.

**Comment by SE2Dev** [comment]
At the time of writing the latest version of the VSCode extension is v0.2.1022 (which was released yesterday). This version still includes `rust-analyzer.inlayHints.enable` in the settings schema - so it probably should be marked as deprecated, or removed entirely.

**Comment by lnicola** [comment]
@SE2Dev see https://github.com/rust-lang/rust-analyzer/pull/12037

**Comment by villem** [comment]
Hi, 
The original problem still prevails. Toggling hints back shows hints after a something has been typed. I'm using rust-analyzer version: 1894473b1 2022-04-23 nightly.

**Comment by lnicola** [comment]
Not a fix for this, but the new version of Code is more flexible in this regard:

Editor › Inlay Hints: Enabled value:

    on - Inlay hints are enabled.
    off - Inlay hints are disabled.
    onUnlessPressed - Inlay hints shown and hidden with Ctrl+Alt.
    offUnlessPressed - Inlay hints hidden and shown with Ctrl+Alt.

**Comment by villem** [comment]
The new "OnUlessPressed" mode works perfectly for me. I'm fine if we close this issue.

**Comment by lnicola** [comment]
I'm worried that this is currently broken anyway, since it toggles between `true` and `false`, not `on` and `off`.

**Comment by villem** [comment]
Yes, vscode preferences editor gave good error about invalid value (true) for inlay hint setting.

**Comment by lnicola** [comment]
@villem actually, can you try the `Toggle inlay hints` command? It seems to work fine for me now, even with the old `true`/`false` settings (so this bug appears to be fixed).

**Comment by lnicola** [comment]
Never mind, it works in a test project, but not in a larger one.

**Comment by lnicola** [comment]
Ah, the dreaded `Content Modified` strikes again.
